### ISSUE TITLE
Questions about model size with auto_train

### ISSUE THREAD

**Issue by xiabo2011** [gitmylo/bark-voice-cloning-HuBERT-quantizer#16]
Hello, this repo is very helpful! But I have a question, when I run the training process with data generated in data-gen. The model file size is over 200M，but the model file size is only around 100M in hugging face: https://huggingface.co/GitMylo/bark-voice-cloning/blob/main/quantifier_hubert_base_ls960.pth. What is the difference between these two files? I think the voice cloned result is pretty good with your hugging face tokenizer model

**Comment by gitmylo** [comment]
the model version 0 is always 100mb, while v1 is 200mb by default.
v1 is known as the large model in [audio webui](/gitmylo/audio-webui). While v0 is the default.

there are 3 models in the [huggingface repo](https://huggingface.co/GitMylo/bark-voice-cloning/tree/main). 2 of version v0 with different amounts of training, and one v1 model with even more training.

**Comment by xiabo2011** [comment]
I got it, does it include the whole training data? https://huggingface.co/datasets/GitMylo/bark-semantic-training/tree/main
I use this data to train v1 model(you mentioned above), the model I got is pretty bad, comparing with your v0 epoch 14 model which is pretty good.

**Comment by gitmylo** [comment]
It was trained on the training data you linked, if that's what you're asking.

How many epochs did you train for? And how did you train it?

**Comment by xiabo2011** [comment]
I tried 14 epochs, but it is v1 model, while your 14 epoch model is v10. If I want to tried to train a v0, how can I modify the model network?

**Comment by gitmylo** [comment]
if you want to train a v0 model, just don't supply version=1 in the quantizer's constructor during training, that way it gets saved as a v0 model. Note that v0 models might have lower quality. And the issues with your training might be due to other reasons. What's the problem exactly?

![image](https://github.com/gitmylo/bark-voice-cloning-HuBERT-quantizer/assets/36931363/71732f79-e456-4878-9ac2-bf0cd0728e78)

**Comment by xiabo2011** [comment]
I see. I'll try again. Thanks a lot

**Comment by xiabo2011** [comment]
Sorry one more question, what was the loss value when you train the model

**Comment by gitmylo** [comment]
The loss is how far the predicted token is from the actual token, using [cross entropy loss](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html)

Cross entropy loss is good for training classification models, which have a lot of outputs, and the highest numbers gets used, this model is also a classification model, it classifies the tokens for Bark based on HuBERT semantic vectors. (technically it works with other wav2vec models too)

**Comment by xiabo2011** [comment]
My fault. I mean what is the loss float value in your training log.

**Comment by gitmylo** [comment]
that's the loss i mentioned before

**Comment by gitmylo** [comment]
just make sure it's going down gradually, it might go up a bit sometimes, that's normal

**Comment by xiabo2011** [comment]
Sorry one more question, what isThroughRandom used for

**Comment by gitmylo** [comment]
ThroughRandom was the old training dataset, where i checked if it would work if i used random semantics, it did not.
